imports/plugins/core/core/server/Reaction/core.js

```diff
@@ -383,6 +384,14 @@ export default {
       shopId = this.getShopIdByDomain();
     }
 
+    // use the primary shop id by default
+    if (!shopId) {
+      const domain = this.connectionDomain();
+      Logger.warn(`No shop matching domain '${domain}'. Defaulting to Primary Shop.`);
+
+      shopId = this.getPrimaryShopId();
```

Comment:
this is the fix, with a logging statement to let us know that our DB is missing data